pkg/geo/geomfn/translate_test.go

```diff
@@ -0,0 +1,122 @@
+// Copyright 2020 The Cockroach Authors.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package geomfn
+
+import (
+	"errors"
+	"testing"
+
+	"github.com/cockroachdb/cockroach/pkg/geo"
+	"github.com/stretchr/testify/require"
+	"github.com/twpayne/go-geom"
+)
+
+func TestTranslate(t *testing.T) {
+	testCases := []struct {
+		desc   string
+		g      geom.T
+		deltas []float64
+	}{
+		{
+			desc:   "move a 2D point",
+			g:      geom.NewPointFlat(geom.XY, []float64{10, 10}),
+			deltas: []float64{5, -5},
+		},
+		{
+			desc:   "move a line string",
+			g:      geom.NewLineStringFlat(geom.XY, []float64{10, 15, 20, 30}),
+			deltas: []float64{1, 5},
+		},
+		{
+			desc:   "move a polygon",
+			g:      geom.NewPolygonFlat(geom.XY, []float64{0, 0, 5, 5, 0, 10, 0, 0}, []int{8}),
```

Comment:
since you've done it, can you add a case for XYZ as well?

Reply:
Aren't 3D objects unsupported? When I add a test case for `XYZ` here, I get an error: `only 2D objects are currently supported` from implicitly calling https://github.com/cockroachdb/cockroach/blob/0b65365fce6a9bdde6c611c92031877ade88a8b2/pkg/geo/geo.go#L621

Reply:
ah i thought this was tested using only the geom.T object. yeah, don't worry about it then.
in that case, can you make expected a fixed geomT, rather than calculating it on the fly in this test case?
you can cast them both to geo.Geometry and compare equality that way -- see any other test in this package for some examples!